client/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,148 @@
+import { useState } from 'react';
+
+import { ResponseUser, RequestUser } from '../../../types/User';
+
+export default function SignUp(): JSX.Element {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  async function registerUser(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const newUser: RequestUser = {
+      username: username,
+      email: email,
+      password: password,
+    };
+    const responseNewUser = await fetch('/api/users', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(newUser),
+    });
+    const dataNewUser: ResponseUser = await responseNewUser.json();
+
+    if (responseNewUser.status === 201) {
+      console.log('User created', dataNewUser);
+    } else {
+      setErrorMessage('Username or email already exists!');
+    }
```

Comment:
This is good! But can be better.

For example, what happens if the server answers with a status code like 500? (Server error)

A better approach would be to check if the response status matches the response status that the server sets when username or email is already used, and then render this error message.

Otherwise, if the status is not 201, we can render a generic error like "sorry, Something went wrong!"